community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/StoreMigrator.java

```diff
@@ -634,6 +714,30 @@ public void moveMigratedFiles( File migrationDir, File storeDir, String versionT
                 true, // allow to skip non existent source files
                 ExistingTargetStrategy.OVERWRITE, // allow to overwrite target files
                 StoreFileType.values() );
+        // Since some of the files might only be accessible through the page cache (i.e. block devices), we also try to
+        // move the files with the page cache.
+        try
+        {
+            Iterable<FileHandle> fileHandles = pageCache.streamFilesRecursive( migrationDir )::iterator;
+            for ( FileHandle fh : fileHandles )
```

Comment:
I was thinking of extracting it only for this class, but a larger refactoring might be a good idea. If you plan to do this, then nevermind, otherwise feel free to address it for this class in this PR.